CRM/Export/BAO/ExportProcessor.php

```diff
@@ -2327,7 +2348,14 @@ public function writeCSVFromTable() {
     // call export hook
     $headerRows = $this->getHeaderRows();
     $exportTempTable = $this->getTemporaryTable();
+    $exportMode = $this->getExportMode();
+    $sqlColumns = $this->getSQLColumns();
+    $componentTable = $this->getComponentTable();
+    $ids = $this->getIds();
     CRM_Utils_Hook::export($exportTempTable, $headerRows, $sqlColumns, $exportMode, $componentTable, $ids);
+    if ($exportMode !== $this->getExportMode() || $componentTable !== $this->getComponentTable()) {
+      CRM_Core_Error::deprecatedFunctionWarning('altering the export mode and/or component table in the hook is no longer supported.');
```

Comment:
@MegaphoneJon so per your discussion  it's kinda meaningless to change ids  & we  are  assuming it's safe  to be  clear about that - shall we  change the hook to no  longer accept $ids  as  a reference? (could be  in master)